lib/teiserver_web/controllers/moderation/report_form_controller.ex

```diff
@@ -47,63 +47,67 @@ defmodule TeiserverWeb.Moderation.ReportFormController do
   @spec create(Plug.Conn.t(), Map.t()) :: Plug.Conn.t()
   def create(conn, %{"report" => report}) do
     target_id = report["target_id"] |> int_parse
-
-    {match_id, relationship} =
-      case report["match_id"] do
-        "none" ->
-          {nil, nil}
-
-        match_id_str ->
-          case Battle.get_match(match_id_str, preload: [:members]) do
-            nil ->
-              {nil, nil}
-
-            match ->
-              target_member =
-                match.members
-                |> Enum.find(fn member -> member.user_id == target_id end)
-
-              reporter_member =
-                match.members
-                |> Enum.find(fn member -> member.user_id == conn.assigns.current_user.id end)
-
-              relationship =
-                cond do
-                  reporter_member == nil -> nil
-                  target_member.team_id == reporter_member.team_id -> "Allies"
-                  target_member.team_id != reporter_member.team_id -> "Opponents"
-                end
-
-              {match.id, relationship}
-          end
+    if conn.assigns.current_user.id == target_id do
+      conn
+      |> put_status(:unprocessable_entity)
```

Comment:
This is a HTML request, if it errors we should be presenting people with a HTML response.

Reply:
I changed the content type and tested that it does now result in an HTML response I didnt know if you wanted a full template or something here, just let me know if you want to see a more sophisticated error page.

Reply:
There should be a moderately user-friendly error page.

Reply:
@Teifion I have added a "self_report" page similar to the existing "no_user" page. let me know if that fits the bill!

Reply:
Thanks :)